src/agent/actions/extract.ts

```diff
@@ -53,13 +74,42 @@ export const ExtractActionDefinition: AgentActionDefinition = {
         );
       }
 
-      const response = await ctx.llm.invoke([
+      const response = await ctx.llm.withStructuredOutput(z.object({variables: VariableFn()})).invoke([
         {
-          role: "user",
+          role: "user", 
           content: [
             {
               type: "text",
-              text: `Extract the following information from the page according to this objective: "${objective}"\n\nPage content:\n${trimmedMarkdown}\nHere is as screenshot of the page:\n`,
+              text: `
+              Original objective (with variable references): "${originalObjective}"
```

Comment:
Let's make this prompt more generic. The focus on the city/country is too specific. Let's also make the examples more varied